evalml/objectives/utils.py

```diff
@@ -30,6 +30,65 @@ def get_non_core_objectives():
     ]
 
 
+def ranking_only_objectives():
+    """Get ranking-only objective classes.
+
+    Ranking-only objectives are objectives that are useful for evaluating the performance of a model, but should not
+    be used as an optimization objective during AutoMLSearch for various reasons.
+
+    Returns:
+        List of ObjectiveBase classes
+    """
+    return [
+        objectives.Recall,
+        objectives.RecallMacro,
+        objectives.RecallMicro,
+        objectives.RecallWeighted,
+        objectives.MAPE,
+        objectives.MeanSquaredLogError,
+        objectives.RootMeanSquaredLogError,
+    ]
+
+
+def get_optimization_objectives(problem_type):
+    """Get objectives for optimization.
+
+    Args:
+        problem_type (str/ProblemTypes): Type of problem
+
+    Returns:
+        List of ObjectiveBase instances
+    """
+    problem_type = handle_problem_types(problem_type)
+    ranking_only = ranking_only_objectives()
+    objectives = [
+        obj
+        for obj in get_ranking_objectives(problem_type)
+        if obj.__class__ not in ranking_only
+    ]
+    return objectives
+
+
+def get_ranking_objectives(problem_type):
+    """Get objectives for pipeline rankings.
+
+    Args:
+        problem_type (str/ProblemTypes): Type of problem
+
+    Returns:
+        List of ObjectiveBase instances
+    """
+    problem_type = handle_problem_types(problem_type)
+    all_objectives_dict = _all_objectives_dict()
+    objectives = [
+        obj()
+        for obj in all_objectives_dict.values()
+        if obj.is_defined_for_problem_type(problem_type)
+        and (obj not in get_non_core_objectives() or obj in ranking_only_objectives())
```

Comment:
remind me what's behind the `obj not in get_non_core_objectives() `

Reply:
There are some objectives in that list that require additional setup/configuration before they can be used during optimization. An example, the cost benefit matrix needs weights for each true positive/negative and false positive/negative